app/src/main/java/com/gusta/wakemehome/services/GeofencingJobIntentService.java

```diff
@@ -0,0 +1,37 @@
+package com.gusta.wakemehome.services;
+
+import android.util.Log;
+
+import androidx.core.app.JobIntentService;
+import androidx.lifecycle.LiveData;
+
+import com.gusta.wakemehome.database.AlarmEntry;
+import com.gusta.wakemehome.database.AppDatabase;
+import com.gusta.wakemehome.viewmodel.MainViewModel;
+
+import java.util.List;
+
+abstract class GeofencingJobIntentService extends JobIntentService {
+
+    // Constant for logging
+    private static final String TAG = MainViewModel.class.getSimpleName();
+
+    protected LiveData<List<AlarmEntry>> mAlarms;
+
+    /**
+     * Override getter to get specific class tag.
+     *
+     * @return Class tag.
+     */
+    abstract protected String getTag();
+
+    /**
+     * Initialize alarms list.
+     */
+    public GeofencingJobIntentService() {
```

Comment:
Because it is LiveData and the "select" runs asynchronously. Both geofencing job intent services needs the current list like the main activity needs it, so I initialize it like there (without the ViewModel cause I read it is bed practice).

Reply:
how did you select here ? didn't it possible only on the main thread?